### ISSUE TITLE
Shaders to perform CPU intensive texture modifications

### ISSUE THREAD

**Issue by Rinnegatamante** [xerpi/sf2dlib#30]
This is probably a stupid question but i'm not familiar very much on how shaders work with sf2dlib so if you can reply me about this it could be really helpful:

I'm in a situation that i need to perform a modification on a texture which is really CPU intensive so i thought it could be a good idea to move this calculation in a shader so that the whole thing should run pretty faster since shader is executed by GPU.

IF you want to take a look, this is the problem: https://github.com/Rinnegatamante/easyrpg-player-3ds/issues/11

At the moment i've suggested to swap to citro3D since it has native support to shaders but maybe integrating some functions to just enable or disable different shaders so that it's possible to swap between the empty shader used by default to someone that should perform some calculation on the texture it's not that hard.

**Comment by xerpi** [comment]
I think we all should switch to citro3d to be honest. sf2dlib uses deprecated libctru GPU functions and primitives. Maybe it would be a good idea to do a thin 2D wrapper to citro3d (citro2d maybe?) with a similar but more complete API than sf2dlib.

**Comment by Rinnegatamante** [comment]
Yeah, that could be REALLY useful since apparently sf2dlib and citro3d are also incompatible (even if you term one of them and then try to init the other, it produces weird behaviours).
